src/Microsoft.Health.Dicom.Core/Features/Query/QueryService.cs

```diff
@@ -44,7 +53,11 @@ public class QueryService : IQueryService
 
             ValidateRequestIdentifiers(message);
 
-            QueryExpression queryExpression = _queryParser.Parse(message);
+            IEnumerable<CustomTagStoreEntry> customTags = await _customTagStore.GetCustomTagsAsync(null, cancellationToken);
+
+            Dictionary<QueryResource, HashSet<CustomTagFilterDetails>> queryResourceToCustomTagMapping = GenerateQueryResourceToCustomTagMapping(customTags);
+
+            QueryExpression queryExpression = _queryParser.Parse(message, queryResourceToCustomTagMapping);
 
```

Comment:
Are the custom tags in the "include" or response tested? Can you make sure it works and add unit tests if not already.

Reply:
It's tested and there are already unit tests covering it!

---
In reply to: [583002754](https://github.com/microsoft/dicom-server/pull/567#discussion_r583002754) [](ancestors = 583002754)